packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -223,47 +223,51 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
     }
   }
 
-  public async closeAllServices(options: { saveTor: boolean } = { saveTor: false }) {
+  public async closeAllServices(
+    options: { saveTor: boolean; purgeLocalDb: boolean } = { saveTor: false, purgeLocalDb: false }
```

Comment:
Feels a little clunky to add another parameter here since `closeAllServices` is doing a little more than just closing things now, but it seems like the simplest change without a larger refactor.

Reply:
[question] Why do we need to add this parameter? Maybe I'm reading into the name `localDbService` too much, but it seems like the data that would be cleared by `this.localDbService.purge()` would be the same data that would be cleared by `this.purgeData()` called in `leaveCommunity()` since that `purgeData()` seems to holistically remove all the local db data from the filesystem. Adding a side effect to `closeAllServices()` seems like it would be a bit of a code smell.

Reply:
I'm not really sure, if we are sure that LevelDB would be totally cleared by `purgeData` then I don't think `this.localDbServer.purge()` is required - I just rearranged what was already there.

Reply:
If we close and then re-open the same LevelDB, is there data in memory that sticks around? Or does removing the LevelDB files clear everything?

Reply:
https://groups.google.com/g/leveldb/c/MXIxcn8akd4?pli=1

Reply:
From the link above "DestroyDB operation will be
faster for moderate to large databases (its implementation works
pretty much as Krzysztof Kowalczyk mentioned: it deletes all the
database files in the directory)." and then taking a look at the source code itself, looks like that is what's going on: https://github.com/google/leveldb/blob/068d5ee1a3ac40dabd00d211d5013af44be55bea/db/db_impl.cc#L1542

Reply:
I just tested this and it appears to work.